File: Data Structures & Algorithms/Arrays/1 Binary Search/BinarySearch.java
import java.util.*;
public class BinarySearch 
{
	private int [] arrA;
	private int number;

	public BinarySearch(int [] arrA)
	{
		this.arrA = arrA;
	}
	public Boolean Search(int low,int high, int number)
	{
		if(low>high)
		{
			return false;
		}
		int mid = low + ((high - low) / 2);
		if(arrA[mid]==number)return true;
		else if (arrA[mid]>number) return Search(low,mid-1,number);
		else return Search(mid+1,high,number);
	}

	public static void main(String args[])
	{	
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter The Size Of The Array : ");
		int n = sc.nextInt();
		int[] a = new int[n];
		System.out.println("Enter The Elements Of The Array : ");
		for(int i=0;i<n;i++)
		{
			a[i] = sc.nextInt();
		}
		System.out.println("Enter The Element To Be Searched : ");
		int s = sc.nextInt();
		BinarySearch sh = new BinarySearch(a);
		System.out.println("The "+ s + " present in array a ??? :" + sh.Search(0, a.length-1, s));

	}

}

File: Data Structures & Algorithms/Arrays/1 Binary Search/Binary Search.c
# include <stdio.h>
int main()
{
	int n;
	printf("Enter The Number Of Elements : ");
	scanf("%d",&n);
	int a[n];
	printf("Enter The Elemets in sorted order:\n");
	int i,s;
	for(int i=0;i<n;i++)
	scanf("%d",&a[i]);
	printf("Enter The Element To Be Searched : ");
	scanf("%d",&s);
	int low,mid,high,flag=0;
	low=0;
	high=n-1;
	mid=(high+low)/2;
	
	while(low<=high)
	{
		
		if(a[mid]==s)
		{
			flag=1;
			break;
		}
		else
		{
			if(s>a[mid])
			{
				low=mid+1;
				mid=(low+high)/2;
			}
			else
			{
				high=mid-1;
				mid=(low+high)/2;
			}
		}
	}	
	

	if(flag==1)
	{
		printf("Element %d Found At %d position",s,mid+1);
	}
	else
	{
		printf("Element Not Found");
	}
}